flocker/acceptance/testtools.py

```diff
@@ -777,10 +785,17 @@ def failed_query(failure):
     # happen know these in advance, but in FLOC-1631 node identification
     # will switch to UUIDs instead.
     agents_connected.addCallback(lambda _: cluster.current_nodes())
+
+    def node_from_dict(node):
+        hostname = node["host"]
+        address = host_mapping.get(hostname, hostname)
```

Comment:
Confusing. Maybe if `host_mapping` was `private_to_public`, and then `public_address = private_to_public.get(hostname, hostname)`.